06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,165 @@
+          this.isValid = {
+            valid: true,
+            message: 'This JSON Format is correct',
+          };
+        } else {
+          this.isValid = {
+            valid: false,
+            message: 'This JSON Format is Incorrect',
+          };
+        }
+      });
+    return this.isValid;
+  }
+
+  /**
+   * A method that create the index of the selected-file
+   * @param  {Object} fileJson the uploaded file object
+   * @param  {Array} uniqueTerms    the unique terms in the JSON file
+   * @param  {String} fileName    The file name
+   * @return {Object} containing the index of the selected-file
+   */
+  createIndex(fileJson, uniqueTerms, fileName) {
+    const indexedDB = {};
+    uniqueTerms.forEach((uniqueKeys) => {
+      const arr = [];
+      fileJson.forEach((jsonObjText) => {
+        arr.push((jsonObjText.text.toLowerCase())
+          .includes(uniqueKeys));
+      });
+      indexedDB[uniqueKeys] = arr;
+    });
+    this.tableObj[fileName] = indexedDB;
+    return indexedDB;
+  }
+
+  /**
+   * A method that gets the index of a filefrom the database object
+   * @param {String} fileName the file name
+   * @return {Object} containing keys of filenames and values of Arrays
+   */
+  getIndex(fileName) {
+    return this.tableObj[fileName];
+  }
+
+
+  /**
+   * A method that searches the index of the current file
+   * @param  {String} query  the string you are currently typing
+   * @param  {String} fileName  the name of the file
+   * @return {Object} A filtered down version of the object you are currently
+   *                  searching, based on the string you are typing.
+   */
+  searchIndex(query, fileName) {
+    if (fileName !== 'all') {
+      const indexedData = this.getIndex(fileName);
+      if (query !== undefined && query.length > 0) {
+        const keyword = this.tokenize(query);
+        this.searchData = {};
+        keyword.forEach((searchKeyUnfiltered) => {
+          const searchKey = searchKeyUnfiltered.toLowerCase();
+          if (searchKey in indexedData) {
+            this.searchData[searchKey] = indexedData[searchKey];
+          }
+        });
+        return this.searchData;
+      }
+      return indexedData;
+    }
+    if (query !== undefined && query.length > 0) {
+      const keyword = this.tokenize(query);
+      const fileNames = Object.keys(this.tableObj);
+      this.searchDataAll = {};
+      fileNames.forEach((allFileName) => {
+        this.searchDataAll[allFileName] = {};
+        keyword.forEach((searchKeyUnfiltered) => {
+          const searchKey = searchKeyUnfiltered.toLowerCase();
+          const filekey = Object.keys(this.tableObj[allFileName]);
+          if (filekey.indexOf(searchKey) > -1) {
+            this.searchDataAll[allFileName][searchKey] =
+            this.tableObj[allFileName][searchKey];
+          }
+        });
+      });
+      return this.searchDataAll;
+    }
+    return this.tableObj;
+  }
+}
+if (typeof window === 'undefined') {
```

Comment:
Thank you for pointing this out, this was the solution i found to an error that kept popping up on my console. i found out It is used to ensure the script is does not run in a web-browser.